backend/models.py

```diff
@@ -54,6 +60,25 @@ class Meta:
         primary_key = CompositeKey('id')
 
 
+class SourceDictionaryLink(Model):
+    # Foreign key to Source with cascade delete
+    source = ForeignKeyField(Source, backref='dictionary_links', to_field='id', on_delete='CASCADE')
+    # Can't be ForeignKeyField due to composite PK in Dictionary
+    dictionary_id = IntegerField()
+    dictionary_timestamp = DateTimeField()
+
+    origin = CharField()  # Options: 'self', 'reused', 'copied', 'imported'
+
+    class Meta:
```

Comment:
Can you set the table name explicitly?
"sources_dictionaries"

Reply:
We’ve decided not to change the table name for now due to a migration issue that we’ll address later